### PR TITLE
Adding Support for chrome extension 'scheme'

### PR DESCRIPTION
**Pull request by leandrocruz**
ZIO Http fails to handle requests from chrome extensions, when the origin header is chrome-extension://EXTENSION_ID.

### CODE REVIEW

File: zio-http/src/main/scala/zio/http/Header.scala [hunk 1]
```diff
@@ -3039,8 +3039,8 @@ object Header {
       if (value == "null") Right(Null)
       else
         URL.decode(value) match {
-          case Left(_)                                              => Left("Invalid Origin header")
-          case Right(url) if url.host.isEmpty || url.scheme.isEmpty => Left("Invalid Origin header")
+          case Left(_)                                              => Left(s"Invalid Origin header '$value'")
```
Comment (leandrocruz) [inline zio-http/src/main/scala/zio/http/Header.scala:3042]: Better error message for clarity

File: zio-http/src/main/scala/zio/http/Scheme.scala [hunk 2]
```diff
@@ -60,14 +62,16 @@ object Scheme       {
...
     def decode(scheme: String)(implicit unsafe: Unsafe): Scheme = {
       if (scheme == null) null
-      else
-        scheme.length match {
-          case 5 => Scheme.HTTPS
-          case 4 => Scheme.HTTP
-          case 3 => Scheme.WSS
-          case 2 => Scheme.WS
-          case _ => null
+      else {
```
Comment (leandrocruz) [inline zio-http/src/main/scala/zio/http/Scheme.scala:65]: Decided for clarity over "speed"

File: zio-http/src/main/scala/zio/http/Scheme.scala [hunk 3]
```diff
@@ -21,10 +21,11 @@ import zio.stacktracer.TracingImplicits.disableAutoTrace
...
   def encode: String = self match {
-    case Scheme.HTTP  => "http"
-    case Scheme.HTTPS => "https"
-    case Scheme.WS    => "ws"
-    case Scheme.WSS   => "wss"
+    case Scheme.HTTP            => "http"
+    case Scheme.HTTPS           => "https"
+    case Scheme.WS              => "ws"
+    case Scheme.WSS             => "wss"
+    case Scheme.ChromeExtension => "chrome-extension"
```
Comment (987Nabil) [inline zio-http/src/main/scala/zio/http/Scheme.scala:28]: I am not sure, that having this explicitly is a good idea. Maybe we should have a `Custom` case, like for http method?
@vigoo @jdegoes

Comment (leandrocruz) [reply zio-http/src/main/scala/zio/http/Scheme.scala:28]: Not sure if this is the right solution also, but, as it is today, zio http fails to accept requests from chrome extensions

**Comment by 987Nabil** [comment]
I think #2489 confirms my hinch. We should have a custom type that stores non predefined schema values.

**Comment by FabioPinheiro** [comment]
I'm just concerned about efficiency.
Is there benchmark that I can run on https://github.com/zio/zio-http/pull/2490?

**Comment by 987Nabil** [comment]
What kind of efficiency are we talking about?
I mean we can match against the ones that have a fixed type and return the fixed type. For all others, we would have 2 options I guess. 
* just put the string as is in a custom wrapper
* validate via regex the validity and only if valid put it in the wrapper

So, where do you see an issue?

**Comment by vigoo** [comment]
I agree to have a `Custom(String)` case instead

**Comment by 987Nabil** [comment]
@vigoo there is https://github.com/zio/zio-http/pull/2490 which is already further in development.
@leandrocruz I am closing this PR in favor of the other.

**Pull request closed without merging**